src/Symfony/Bridge/Doctrine/Middleware/ConnectionKeepAlive.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Connection as DriverConnection;
+use Doctrine\DBAL\Driver\Middleware;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use Symfony\Bridge\Doctrine\ConnectionLossAwareHandler;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Based on https://github.com/Baldinof/roadrunner-bundle/blob/3.x/src/Integration/Doctrine/DoctrineORMMiddleware.php.
+ */
+class ConnectionKeepAlive implements Middleware
+{
+    public function __construct(private ManagerRegistry $managerRegistry, private ContainerInterface $container)
+    {
+    }
+
+    public function wrap(Driver $driver): Driver
+    {
+        return new class($driver, $this->managerRegistry, $this->container) extends AbstractDriverMiddleware {
+            public function __construct(private Driver $driver, private readonly ManagerRegistry $managerRegistry, private readonly ContainerInterface $container)
+            {
+                parent::__construct($driver);
+            }
+
+            public function connect(array $params): DriverConnection
+            {
+                $connectionServices = $this->managerRegistry->getConnectionNames();
+
+                foreach ($connectionServices as $connectionServiceName) {
```

Comment:
having a middleware for one connection that reconnects all connections is a no-go for me. It does not make any sense. A middleware should apply only to the connection it is applied on.

Reply:
Btw, this listener will actually apply on all connections **except** the one being wrapped by the middleware (because by the time the `->connect()` middleware is called, the current connection is not connected yet). And it would be called after the connection has been established earlier and lost (so it won't solve the case of long running runtime AFAICT)

Reply:
Thank you for your feedback.
I'm a bit confused. How could I be aware of the currently used connection for the requested operation? Also, Are you suggesting that it's not through this type of middleware that the result can be achieved in terms of when it will be called?

I appreciate your explanations in advance 🙏

Reply:
actually, as discussed, this middleware already wraps the connection so maybe, it would be more appropriate to consider a different approach